### PR TITLE
feat(bin, pipeline): report pipeline progress

### PR DESCRIPTION
**Pull request by shekhirin**
During the initial sync, it's useful to know not only the progress of each individual stage in the pipeline, but also how many stages left. 

```console
2023-05-31T14:40:41.012518Z  INFO Stage committed progress pipeline_stages=3/13 stage=Bodies progress=3230000 checkpoint=3230000
```

### CODE REVIEW

**Comment by mattsse** [comment]
> stage=Bodies pipeline_progress=3/13

what if we do:

```
stage=Bodies (3/13)
```

**Comment by shekhirin** [comment]
@mattsse thought about it, I think it might worsen the UX of searching through logs if you load them into some system like journald and want to search for the logs of specific stage.

**Comment by mattsse** [comment]
I see, 
I'd suggest renaming `pipeline_progress` to `stage=3/13` or something similar, so we don't have progress twice in there.

**Comment by shekhirin** [comment]
@mattsse renamed to `pipeline_stages`, `stage` is already taken by stage name. Lmk if you have a better naming idea.

File: bin/reth/src/node/events.rs [hunk 1]
```diff
@@ -37,22 +37,28 @@ impl NodeState {
...
-            PipelineEvent::Running { stage_id, checkpoint } => {
+            PipelineEvent::Running { pipeline_progress, pipeline_total, stage_id, checkpoint } => {
                 let notable = self.current_stage.is_none();
                 self.current_stage = Some(stage_id);
                 self.current_checkpoint = checkpoint.unwrap_or_default();
 
                 if notable {
                     info!(
                         target: "reth::cli",
+                        pipeline_stages = %format!("{pipeline_progress}/{pipeline_total}"),
```
Comment (mattsse) [inline bin/reth/src/node/events.rs:48]: this makes sense to me,

I'm unsure about the order here, but what we have:

`stages=y/x name progress checkpoint`

is the best order imo

File: crates/stages/src/pipeline/event.rs [hunk 2]
```diff
@@ -12,13 +12,21 @@ use reth_primitives::stage::{StageCheckpoint, StageId};
 pub enum PipelineEvent {
     /// Emitted when a stage is about to be run.
     Running {
+        /// 1-indexed ID of the stage that is about to be run out of total stages in the pipeline.
+        pipeline_progress: usize,
```
Comment (mattsse) [inline crates/stages/src/pipeline/event.rs:16]: personally, I'd rather have `pipeline_index` or `pipeline_position` here

**Review by mattsse — COMMENTED** [review]
supportive of this.

pending wdyt @onbjerg

**Review by mattsse — APPROVED** [review]
lgtm,

pending @rkrasiuk or @onbjerg

**Review by onbjerg — APPROVED** [review]
lgtm

**Pull request merged**